### DISCUSSION TITLE
Milvus Clustering on 2P Systems

### DISCUSSION THREAD

**Discussion by msenizai** [milvus-io/milvus#33740] [Q&A and General discussion]
We started testing VendorDBBench on Milvus and noticed that the performance drops when using 2P systems vs their 1P counterparts.  Since the database is a single process with many threads this is a common occurance.

Milvus Standalone v4.2.0
64 Cores on  1 socket 500qps, 
64 Cores on 2 socket 350qps.

Also noticed that when this runs only about 50% of the allocated CPU is running, meaning some sub-components are getting cores allocated but not used during the long running phases.  Regardless of the pinning strategy.

Looking at Milvus Cluster, is there any guidance on how many resources to provide to a given container so it isn't constrained, and have high efficiency by not crossing sockets for critical work or leaving too many cores idle?

I did a quick test with 2 data nodes, binding with 'cpuset' to keep cores on individual sockets, but performance dropped a little further.  

Don't have a lot of systems for testing, but this will help inform our placement strategy moving forward so we get the highest level of scale up and scale out efficiency.

**Comment by xiaofan-luan** [comment]
I don't get enough details about hardware you are using, but there seems to be similar issue as NUMA architecture, which means copy and access data becomes slower when across socket.
To solve that, limit your pod to 32 core or 16 core and make sure each pod locate on one machine, you can definite change your helm chart or operator to limit all node.
And you should also check the host machine cpu usage and see if host cpu is more than 70%. If it is still small this usually means you have a room to improve
